docs/fundamentals/code-analysis/style-rules/ide0160-ide0161.md

```diff
@@ -0,0 +1,109 @@
+---
+title: "Namespace declaration preferences (IDE0160 and IDE0161)"
+description: "Learn about code analysis rules IDE0160 and IDE0161: Namespace declaration preferences"
+ms.date: 06/28/2022
+ms.topic: reference
+f1_keywords:
+- IDE0160
+- IDE0161
+- csharp_style_namespace_declarations
+helpviewer_keywords:
+- IDE0160
+- IDE0161
+- csharp_style_namespace_declarations
+dev_langs:
+- CSharp
+---
+# Namespace declaration preferences (IDE0160 and IDE0161)
+
+This article describes two related rules, `IDE0160` and `IDE0161`.
+
+| Property                 | Value                                  |
+| ------------------------ | -------------------------------------- |
+| **Rule ID**              | IDE0160                                |
+| **Title**                | Use block-scoped namespace             |
+| **Category**             | Style                                  |
+| **Subcategory**          | Language rules (namespace preferences) |
+| **Applicable languages** | C#                                     |
+| **Options**              | `csharp_style_namespace_declarations`  |
+
+| Property                 | Value                                  |
+| ------------------------ | -------------------------------------- |
+| **Rule ID**              | IDE0161                                |
+| **Title**                | Use file-scoped namespace              |
+| **Category**             | Style                                  |
+| **Subcategory**          | Language rules (namespace preferences) |
+| **Applicable languages** | C#                                     |
+| **Options**              | `csharp_style_namespace_declarations`  |
+
+## Overview
+
+These rules concern how namespaces are declared. For `IDE0161` to report violations when block-scoped namespaces are used, you must set the associated option to `file_scoped`.
```

Comment:
As stated in https://github.com/dotnet/roslyn/issues/60546#issuecomment-1086854845, it's unclear why this option is even necessary. And what happens if you set it to `file_scoped` and enable IDE0160 (use block scoped namespaces)?

Reply:
The option is used when creating a new file in Visual Studio to determine whether the new file uses file scoped or not. This is true even if both IDE0160 and IDE0161 are disabled.